db/db_impl/db_impl.h

```diff
@@ -852,8 +852,8 @@ class DBImpl : public DB {
   InstrumentedMutex* mutex() const { return &mutex_; }
 
   // Initialize a brand new DB. The DB directory is expected to be empty before
-  // calling it.
-  Status NewDB();
+  // calling it. Optionally return new manifest file name.
+  Status NewDB(std::vector<std::string>* new_filenames = nullptr);
```

Comment:
There are other usages of NewDB() (e.g. repair.cc:188)

Reply:
Yeah, this is the only other case.

Reply:
> Yeah, this is the only other case.

I feel hesitant to modify the other use case, as the "nullptr" as argument makes less sense there. Please let me know if that's actually better.

Reply:
Declaring a function with default parameter has one advantage: it makes you write less code. This comes at the cost of less readability: when you see `NewDB()`, you need to check its declaration to figure out what the signature is.
In this case, the benefit of less code is minimal since it is used only twice in the entire codebase. The cons outweigh the pros.
Passing a nullptr means we do not need return value. It still makes sense: sometimes we need a out-arg as return value, sometimes we do not.

Reply:
Plus, writing it as `NewDB(/*new_filenames=*/nullptr)` is preferred. We have been following this convention to further improve readability.

Reply:
Super clear! Thanks!